### PR TITLE
Support for JAVA

### PR DESCRIPTION
**Pull request by sushilraje**
# Description and Motivation 

...

# Change type 

- [ ] Bug fix
- [x] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

**Checklist:**

- [ ] All tests passed
- [ ] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: ASAManager/run.sh [hunk 1]
```diff
@@ -52,14 +52,65 @@ run_tests() {
     docker logs "asa-manager"
 }
 
-header "Running $TEST_SUITE"
 
-check_dependency_docker
-check_dependency_dotnet
+#Script options  
+repo=""
```
Comment (ppathan) [inline ASAManager/run.sh]: ASA Manager is only dotnet. So that should be default and should ignore java.

File: ASAManager/run.sh [hunk 2]
```diff
@@ -52,14 +52,65 @@ run_tests() {
...
+if [ "$repo" == "both" ] || [ "$repo" == "" ]; then
+	export REPO=dotnet
+	run
+	
+	export REPO=java
+	run
+elif [[ "$repo" == "dotnet" ]]; then
+	export REPO=dotnet
+	run
+elif [[ "$repo" == "java" ]]; then
```
Comment (ppathan) [inline ASAManager/run.sh]: Remove java

File: ASAManager/run.sh [hunk 3]
```diff
@@ -52,14 +52,65 @@ run_tests() {
...
+tag="testing"
+
+set_up() {
+    export DOCKER_TAG=$tag
+	check_dependency_docker
+	check_dependency_dotnet
+}
+
+run() {
+	start_containers
```
Comment (ppathan) [inline ASAManager/run.sh]: Tab size seems off. It should be 4 but this looks like 8.

File: ASAManager/run.sh [hunk 4]
```diff
@@ -52,14 +52,65 @@ run_tests() {
...
 
-check_dependency_docker
-check_dependency_dotnet
+#Script options  
+repo=""
+tag="testing"
+
+set_up() {
+    export DOCKER_TAG=$tag
+	check_dependency_docker
```
Comment (ppathan) [inline ASAManager/run.sh]: nit: indentation

File: scripts/storageadapter.sh [hunk 5]
```diff
@@ -51,6 +51,7 @@ create_network() {
 }
 
 start() {
+ 	echo "Sushil"$DOCKER_IMAGE
```
Comment (ppathan) [inline scripts/storageadapter.sh]: Remove

File: ASAManager/run.sh (hunk 4 above)
Comment (sushilraje) [reply ASAManager/run.sh]: Yes, I m working on it.

File: scripts/build [hunk 6]
```diff
@@ -8,20 +8,48 @@ cd $APP_HOME
...
+    ./StorageAdapter/run.sh -re $repo -t $tag
 
-./IoTHubManager/run.sh "IoT Hub Manager tests"
+    ./Config/run.sh -re $repo -t $tag
 
-./ASAManager/run.sh "ASA Manager tests"
+    ./IoTHubManager/run.sh -re $repo -t $tag
 
-./Telemetry/run.sh "Telemetry tests"
+	./ASAManager/run.sh -re $repo -t $tag
```
Comment (ppathan) [inline scripts/build]: nit: wrong extra tab

File: scripts/build [hunk 7]
```diff
@@ -8,20 +8,48 @@ cd $APP_HOME
...
+
+run_with_options() {
+    if [[ "$repo" == "both" ]]; then
+        header2 "Running both .Net and Java tests"
+	repo=dotnet
+	run_tests 
+	repo=java
+	#run_tests
+    else
+	run_tests
```
Comment (ppathan) [inline scripts/build]: will this only run dotnet tests?

File: scripts/build [hunk 8]
```diff
@@ -8,20 +8,48 @@ cd $APP_HOME
...
+    ./Telemetry/run.sh -re $repo -t $tag
+}
+
+run_with_options() {
+    if [[ "$repo" == "both" ]]; then
+    	header2 "Running both .Net and Java tests"
+	repo=dotnet
+	run_tests 
+	repo=java
+	#run_tests
```
Comment (ppathan) [inline scripts/build]: Wanted to make sure that we are starting by first disabling the java tests, is that true?

File: scripts/build (hunk 7 above)
Comment (sushilraje) [reply scripts/build]: Yes if NO arguments supplied run only .Net tests will be run.
But If anyone wants java (to debug failures) then they can 
run by ./build -re java

**Pull request merged**